src/main/scala/diplomacy/SRAM.scala

```diff
@@ -34,16 +32,43 @@ abstract class DiplomaticSRAM(
   def mask: List[Boolean] = bigBits(address.mask >> log2Ceil(beatBytes))
 
   // Use single-ported memory with byte-write enable
-  def makeSinglePortedByteWriteSeqMem(size: BigInt, lanes: Int = beatBytes, bits: Int = 8) = {
+  def makeSinglePortedByteWriteSeqMem(
+    size: BigInt,
+    lanes: Int = beatBytes,
+    bits: Int = 8,
+    busProtocol: Option[OMProtocol],
+    dataECC: Option[OMECC] = None,
+    hasAtomics: Option[Boolean] = None) = {
     // We require the address range to include an entire beat (for the write mask)
     val mem =  DescribedSRAM(
       name = devName.getOrElse("mem"),
       desc = devName.getOrElse("mem"),
       size = size,
-      data = Vec(lanes, UInt(width = bits))
+      data = Vec(lanes, UInt(width = bits)),
+      id = DescribedSRAMIdAssigner.genId()
     )
     devName.foreach(n => mem.suggestName(n.split("-").last))
 
+    val omSRAM: OMSRAM = DiplomaticObjectModelAddressing.makeOMSRAM(
+      desc = "mem", //lim._2.name.map(n => n).getOrElse(lim._1.name),
+      depth = size,
+      data = Vec(lanes, UInt(width = bits)),
+      id = DescribedSRAMIdAssigner.genId()
```

Comment:
Is it correct to call this function twice? Shouldn't you call it once and use the result in both places?

Reply:
Fixed